src/templates/macros/components.html

```
@@ -0,0 +1,54 @@
+{#- Reusable UI component macros with built-in accessibility attributes.
+    Import with: {% from 'macros/components.html' import button, form_field, modal, status_chip, card %}
+-#}
+
+{% macro button(text, type="button", variant="primary", icon=None, disabled=False, attrs={}) -%}
+<button type="{{ type }}" class="action-button{% if variant != 'primary' %} {{ variant }}{% endif %}"{% if disabled %} disabled{% endif %}{% for key, val in attrs.items() %} {{ key }}="{{ val }}"{% endfor %}>
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ < ].

(spec-char-escape)